### DISCUSSION TITLE
Plans for Updating SNP Packages to Align with Latest Tesseract Updates.

### DISCUSSION THREAD

**Discussion by aizmailovs** [ros-industrial-consortium/scan_n_plan_workshop#147] [General]
Hi all,

Recently, I have decided to try out the [program formatting task recently developed in tesseract_planning](https://github.com/tesseract-robotics/tesseract_planning/issues/454) and integrate it into scan-n-plan workshop. However, updating the tesseract packages caused build issues with the snp packages, likely due to dependency on earlier tesseract versions.

Are there plans for the snp package to be updated in the nearest future to align with the latest updates in tesseract?

**Comment by marip8** [comment]
I started an update to Tesseract 0.23 [here](https://github.com/ros-industrial-consortium/scan_n_plan_workshop/pull/137) but I haven't had a chance to finish it yet. This version of Tesseract won't include the program formatting task, but it might get you a long ways towards building this repository with a version of Tesseract that does include it

**Reply by aizmailovs** [reply]
Thank you for your reply @marip8! I will take a look into this update.

**Reply by aizmailovs** [reply]
I fetched the PR, and SNP was able to build together with the Tesseract 0.26. So, I am good to go.

**Reply by marip8** [reply]
Good to hear. I plan to finish the update soon. I think the last outstanding component of that PR is cleaning up the planning pipeline YAML file. I'll try to bump it out to Tesseract 0.26 as well afterwards

**Reply by aizmailovs** [reply]
That is good news! Yesterday, I also came to the same conclusion, assuming we are referring to the same file (`task_composer_plugins.yaml` within snp_motion_planning). I was planning to continue working on it next week.

> I plan to finish the update soon.

Would you recommend that I wait for the update, or should I update the pipeline and present the outcome next week?
